src/engine.cairo

```diff
@@ -164,8 +168,54 @@ pub impl BlockHeaderEngineImpl of BlockHeaderEngineTrait {
 }
 
 // Helper functions
-fn bits_to_target(bits: u32) -> u256 {
-    0
+pub fn bits_to_target(bits: u32) -> Result<u256, felt252> {
```

Comment:
Implementation here: https://github.com/ZeroSync/ZeroSync/blob/f1bd4736b0bac190ac48cbc8d5fcd92e2e1be7b9/src/block/block_header.cairo#L178 looks much simpler. It seems there is no need to use loops.